src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -248,8 +255,18 @@ abstract class ManagedStrategy(
     protected fun runInvocation(): InvocationResult {
         initializeInvocation()
         val result = runner.run()
+        // In case when a runner detects a deadlock, some threads can still work with current strategy
```

Comment:
Please remove "when" and replace "a" with "the" and "work with" with "manipulate the"